pkg/irsa/irsa.go

```diff
@@ -195,19 +262,19 @@ func (s *IRSAService) Delete(ctx context.Context) error {
 	if err != nil {
 		ctrlmetrics.Errors.WithLabelValues(s.Scope.Installation(), s.Scope.AccountID(), s.Scope.ClusterName(), s.Scope.ClusterNamespace()).Inc()
 		s.Scope.Logger.Error(err, "failed to delete S3 files")
-		return microerror.Mask(err)
+		return err
 	}
 	err = s.S3.DeleteBucket(s.Scope.BucketName())
 	if err != nil {
 		ctrlmetrics.Errors.WithLabelValues(s.Scope.Installation(), s.Scope.AccountID(), s.Scope.ClusterName(), s.Scope.ClusterNamespace()).Inc()
 		s.Scope.Logger.Error(err, "failed to delete S3 bucket")
-		return microerror.Mask(err)
+		return err
 	}
 	err = s.IAM.DeleteOIDCProvider(s.Scope.AccountID(), s.Scope.BucketName(), s.Scope.Region())
 	if err != nil {
 		ctrlmetrics.Errors.WithLabelValues(s.Scope.Installation(), s.Scope.AccountID(), s.Scope.ClusterName(), s.Scope.ClusterNamespace()).Inc()
 		s.Scope.Logger.Error(err, "failed to delete OIDC provider")
-		return microerror.Mask(err)
```

Comment:
Why don't we mask any more?

Reply:
The error will be masked in controller.go